app/questionnaire/path_finder.py

```diff
@@ -59,19 +59,32 @@ def build_path(self):
                 group_blocks = self._build_blocks_for_group(group, instance_idx)
                 blocks += group_blocks
 
-            if group['id'] in first_groups:
-                this_location = Location(group['id'], 0, group['blocks'][0]['id'])
+            first_group_instance = self._get_first_group_instance_in_block(group, blocks)
+            if first_group_instance is not None:
+                if group['id'] in first_groups:
+                    this_location = Location(group['id'], first_group_instance, group['blocks'][0]['id'])
```

Comment:
This might be clearer if you just set the location's group_instance rather than make a new location? (unless there's a reason I'm missing!) e.g. `this_location.group_instance = first_group_instance`

Reply:
see above

Reply:
I think this point is unrelated to the above one